llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -253,6 +254,40 @@ void VPRecipeBase::moveBefore(VPBasicBlock &BB,
   insertBefore(BB, I);
 }
 
+InstructionCost VPRecipeBase::cost(ElementCount VF, VPCostContext &Ctx) {
+  if (auto *S = dyn_cast<VPSingleDefRecipe>(this)) {
+    auto *UI = dyn_cast_or_null<Instruction>(S->getUnderlyingValue());
```

Comment:
Note that other types of recipes associated with underlying instructions (dealt with below in VPRecipeBase::computeCost()), namely interleave-group and widen-memory, are irrelevant for skipping cost.

Reply:
👍🏻